### DISCUSSION TITLE
Pass parameters through IRegionNavigationJournal.GoBack()

### DISCUSSION THREAD

**Discussion by achobanov** [PrismLibrary/Prism#2553] [WPF] [answered]
I know it might seem counter-productive for a `GoBack` implementation to allow adding additional parameters, but it is exactly what I want to do in my case. 

I am building an UI to configure a set of tree structures. The tree roots are listed in the main page, selecting one brings up the root node configuration in view. However the tree itself is not visualized in the UI and each child node , much like the root, represents a new View-ViewModel pair. The currently active node occupies the **Content** region, while others are not visible. 

What I am looking for now is a clean way to add new child configuration entries. For example: an **Event** can have many **Competitions**. When I add a new competition, Prism brings out a new view, but I have not yet found a good way to add the new instance to the parent. I tried using:
1. Events - it's not a good fit, since what I want to do is instance-specific. I.e. to make it work - in each node ViewModel I would have to subscribe and then filter by some node-unique value. Otherwise adding a child will result in all instances of it's parent to have the same child reference
2. Reflection - I came up with a complicated solution using Reflection in a base-class to pass an Action<T> parameter to the child node, which the child calls when it's configuration is done, using itself as a parameter, which action essentially adds the child reference in the parent's children collection. All that happens automatically and is as compile-time safe as possible. That said - it is far from ideal, because of it's complexity. 

So here I am - if I could pass parameters to the `GoBack` method, I would be able to override the behavior of child ViewModels to pass themselves as a navigation parameter and then simply add them in the parent. Simple, safe and maintainable. I've setup my application's UI simply, only one region changes, which means that a child will always backup to it's exact parent. All I need now is a way to pass the data along.

**Comment by brianlagunas** [comment, marked as answer]
There two common approaches to this; the event aggregator, or add the parameters via the OnNavigatedFrom method and capture them in the OnNavigatedTo.

**Reply by achobanov** [reply]
Hi, Brian, thanks for the quick reply. Also for your work on the framework - so far it has been awesome. Do I understand correctly that if I add the parameter in the child's `OnNavigatedFrom`, it's parent will receive it, even when the navigation happens through the `Journal.GoBack` method? I'll try it out momentarily.

**Reply by brianlagunas** [reply]
Not sure what you mean by the "parent", but the target of the goback operation (the previous page) will receive the parameters. You can also use RegionContext if you need to share data between the view hosting the region and any views in the region.

See: https://prismlibrary.com/docs/wpf/legacy/Communication.html#region-context

**Reply by achobanov** [reply]
Parent as in parent node from my example above. That'll do the trick. Thanks a lot!
